internal/cli/root/builder.go

```diff
@@ -289,43 +253,6 @@ func shouldSetService(cmd *cobra.Command) bool {
 	return true
 }
 
-func shouldCheckCredentials(cmd *cobra.Command) AuthRequirements {
-	searchByName := []string{
-		"__complete",
-		"help",
-	}
-	for _, n := range searchByName {
-		if cmd.Name() == n {
-			return NoAuth
-		}
-	}
-	customRequirements := map[string]AuthRequirements{
-		fmt.Sprintf("%s %s", atlas, "completion"):  NoAuth,       // completion commands do not require credentials
-		fmt.Sprintf("%s %s", atlas, "config"):      NoAuth,       // user wants to set credentials
-		fmt.Sprintf("%s %s", atlas, "auth"):        NoAuth,       // user wants to set credentials
-		fmt.Sprintf("%s %s", atlas, "register"):    NoAuth,       // user wants to set credentials
-		fmt.Sprintf("%s %s", atlas, "login"):       NoAuth,       // user wants to set credentials
-		fmt.Sprintf("%s %s", atlas, "logout"):      NoAuth,       // user wants to set credentials
-		fmt.Sprintf("%s %s", atlas, "whoami"):      NoAuth,       // user wants to set credentials
-		fmt.Sprintf("%s %s", atlas, "setup"):       NoAuth,       // user wants to set credentials
-		fmt.Sprintf("%s %s", atlas, "register"):    NoAuth,       // user wants to set credentials
-		fmt.Sprintf("%s %s", atlas, "plugin"):      NoAuth,       // plugin functionality requires no authentication
-		fmt.Sprintf("%s %s", atlas, "quickstart"):  NoAuth,       // command supports login
-		fmt.Sprintf("%s %s", atlas, "deployments"): OptionalAuth, // command supports local and Atlas
-	}
-	for p, r := range customRequirements {
-		if strings.HasPrefix(cmd.CommandPath(), p) {
-			return r
-		}
-	}
-
-	if plugin.IsPluginCmd(cmd) || pluginCmd.IsFirstClassPluginCmd(cmd) {
-		return OptionalAuth
-	}
```

Comment:
[qq] how will this work for plugins? now that we moved it under transport, maybe it won't show up for K8s anymore - but since it's optional auth it's probably all right